pkg/asset/installconfig/aws/validation.go

```diff
@@ -329,12 +330,28 @@ func validateEndpointAccessibility(endpointURL string) error {
 	if port == "" {
 		port = "https"
 	}
-	conn, err := proxy.Dial(context.Background(), "tcp", net.JoinHostPort(URL.Hostname(), port))
-	if err != nil {
-		return err
+	err = proxyDialURL(context.Background(), net.JoinHostPort(URL.Hostname(), port))
+	return err
+}
+
+func proxyDialURL(ctx context.Context, URL string) error {
+	conn, err := proxy.Dial(context.Background(), "tcp", URL)
+	if err == nil {
+		conn.Close()
+		return nil
+	}
+	// DefaultClient picks the http and https proxy environment variables.
+	req, err := http.NewRequestWithContext(context.Background(), "CONNECT", URL, nil)
```

Comment:
Why `CONNECT`? Can we replace this with a simple `http.Get`?

Reply:
I'm trying to relearn everything about http requests but AFAIK GET will request for resources that may not be available and CONNECT just gets the connection to the server?

Reply:
CONNECT is used to tunnel through a proxy. If you are going to use CONNECT, then the URL would need to be the proxy URL and you would need to build out the request to have the connection routed to the service endpoint.

Reply:
I think the http request function expects a URL and automatically picks the proxy URL from the environment variables to connect to the service endpoint and we just need to provide the service endpoint here.

Reply:
I don't think that is what is going to happen here. There is no sense in making a CONNECT without subsequent http requests through the tunnel. The http.Request does not support using the tunnel.

Reply:
The client--not the request--will pick up the proxy from the environment variables. To reiterate staebler's point CONNECT is used to connect to a proxy, but right now the code is trying to connect to the AWS endpoint as if the endpoint were the proxy.

I haven't heard any objections so I think we can still greatly simplify this down to just `http.Get`

Reply:
Got it. I changed it to GET now.